rules/klayout/drc/testing/run_regression.py

```diff
@@ -32,28 +32,30 @@
     --run_name=<run_name>               Select your run name.
 """
 
+from subprocess import check_call
+
 from docopt import docopt
 import os
 import datetime
 import xml.etree.ElementTree as ET
 import csv
 import time
 import re
-
-from sympy import arg
+import pandas as pd
+import logging
 
 def call_regression(rule_deck_path, path):
+
     t0 = time.time()
     marker_gen = []
-    rules =[]
-    ly = 0
+    rules      = []
+    ly         = 0
 
     # set folder structure for each run
     x = f"{datetime.datetime.now()}"
     x = x.replace(" ", "_")
-
     name_ext = str(rule_deck_path).replace(".drc","").split("/")[-1]
-    os.system(f"mkdir run_{x}_{name_ext}")
+    check_call(f"mkdir run_{x}_{name_ext}", shell=True)
 
     # Get the same rule deck with gds output
     with open(rule_deck_path, 'r') as f:
```

Comment:
is this generating a file? curious if you've considering using a templating system like https://jinja.palletsprojects.com/ or a multi-line format string? That could help convey a better understanding of the final shape of the file to the reader.

Reply:
@proppy No it's generating an output folder structure for testing.

Reply:
It seems to do a bit more than that: it reads `rule_deck_path`,  then rewrite some of the line before appending them to  ` marker_gen` and then write the resulting strings to a file named `markers.drc`.

I'm curious why that rewriting is necessary (maybe add more comments), and if there is a better way (template?) to process this.

Reply:
@proppy Yes, it generates a new rule deck from the original one and change the output to GDS layer rather than database output. For regression purposes.

Reply:
can't this be a parameter passed to the rule deck using `-rd` rather than rewriting it?

Reply:
@proppy No, that's a complicated thing to do. Also, the original rule deck doesn't need this output. It's only for regression purposes.